app/controllers/users/phone_setup_controller.rb

```diff
@@ -135,5 +139,30 @@ def new_phone_form_params
         :recaptcha_mock_score,
       )
     end
+
+    def check_phone_submission_limit(phone)
+      return false if phone.blank?
+      fingerprint = Pii::Fingerprinter.fingerprint(Phonelib.parse(phone).e164.to_s)
+      @submission_rate_limiter ||= RateLimiter.new(
+        target: fingerprint,
+        rate_limit_type: :phone_fingerprint_confirmations,
+      )
+      @submission_rate_limiter.increment!
+      lock_out_user if @submission_rate_limiter.maxed?
+    end
+
+    def lock_out_user(user: current_user)
+      UpdateUser.new(user: user, attributes: { second_factor_locked_at: Time.zone.now }).call
```

Comment:
Do we need this? I don't know that we want to use the `second_factor_locked_at` mechanism, since that locks out the account entirely for use with signing in. It should be enough to rely on the rate limiter to prevent further submissions in the same time period.

```suggestion
```

Reply:
That is now removed.